tools/build-flow-types.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+/* eslint-disable no-console, import/no-commonjs */
+/**
+ * Build flow type files for exports.
+ *
+ * Generates *.js.flow files in dist for each file that is an export for
+ * a given package. It means we have to ship source code but we're doing that
+ * anyway right now, so no worries.
+ *
+ * Besides index.js files, we also export some other specific files for
+ * general consumption. In order to make this scalable, but reasonably fast
+ * we assume that each index.js file has a header in it that identifies
+ * additional export files. That way we only need to look for index.js files
+ * and read those to work out the rest. This makes it at least a little
+ * scalable but also self-documenting.
+ */
+const {promisify} = require("util");
+const fs = require("fs");
+const path = require("path");
+const ancesdir = require("ancesdir");
+
+/**
+ * Let's use async/await to make life nicer.
+ */
+const readFileAsync = promisify(fs.readFile);
+const writeFileAsync = promisify(fs.writeFile);
+
+/**
+ * Some regex's we use to support our additional-files syntax.
+ */
+/**
+ * The first extracts the @additional-exports block, which looks like this:
+ *
```

Comment:
style nit: these comment blocks could be joined.

Reply:
I kept them separate so that hopefully, the intellisense doesn't include the bit that's only for documenting locally in the file